crates/ruff_python_formatter/src/statement/suite.rs

```diff
@@ -472,15 +472,28 @@ fn stub_file_empty_lines(
         || !stub_suite_can_omit_empty_line(preceding, following, f);
     match kind {
         SuiteKind::TopLevel => {
-            if empty_line_condition {
+            if empty_line_condition
+                || (is_blank_line_after_nested_stub_class_enabled(f.context())
+                    && blank_line_after_nested_stub_class_condition(
+                        preceding.into(),
+                        Some(following.into()),
+                        f,
+                    ))
+            {
                 empty_line().fmt(f)
             } else {
                 hard_line_break().fmt(f)
             }
         }
         SuiteKind::Class | SuiteKind::Other | SuiteKind::Function => {
-            if empty_line_condition
-                && lines_after_ignoring_end_of_line_trivia(preceding.end(), source) > 1
+            if (empty_line_condition
+                && lines_after_ignoring_end_of_line_trivia(preceding.end(), source) > 1)
+                || (is_blank_line_after_nested_stub_class_enabled(f.context())
+                    && blank_line_after_nested_stub_class_condition(
+                        preceding.into(),
+                        Some(following.into()),
+                        f,
+                    ))
```

Comment:
It's difficult to merge the `is_*_enabled && *_condition` logic with `empty_line_condition` variable up above like so:

```rust
    let empty_line_condition = preceding_comments.has_trailing()
        || following_comments.has_leading()
        || !stub_suite_can_omit_empty_line(preceding, following, f)
        || (is_blank_line_after_nested_stub_class_enabled(f.context())
                    && blank_line_after_nested_stub_class_condition(
                        preceding.into(),
                        Some(following.into()),
                        f,
                    ))
```

The reason being that in the top level condition the boolean check is direct but in other suite kind there's an additional check of the number of empty lines. That additional check is to _only_ preserve the empty lines if they're present while the preview style adds an empty line conditionally. So, the condition should OR the complete check which was existing earlier to complement it.

Reply:
I think what would help here is to rename `empty_line_condition` into `preserve_empty_line` which is different from your check which forces empty lines. 

I recommend moving the `blank_line_after_nested_stub_class_condition` out of the `match` and next to `empty_line_condition`, and name the variable `require_empty_line`

I would rename `blank_line_after_nested_stub_class_condition` to `requires_empty_line_between_stub_file_statements` (not just preserving, enforcing an empty line).

We can then rewrite the checks in the match branches to 

```
if requires_empty_line || preserve_empty_line {
    empty_line.fmt(f)
}
```

which I find easier to understand.

Reply:
> I think what would help here is to rename `empty_line_condition` into `preserve_empty_line` which is different from your check which forces empty lines.

I'm not sure if this is 100% correct. For top-level, if the condition is true, we always make sure that there is just one empty line. It could be that there isn't any empty line in the source code which means we're adding an additional empty line. But, for the nested cases, we'd only preserve any existing empty lines and limit it to 1.

> I recommend moving the `blank_line_after_nested_stub_class_condition` out of the `match` and next to `empty_line_condition`, and name the variable `require_empty_line`

Thanks, I'll update the code accordingly.

> I would rename `blank_line_after_nested_stub_class_condition` to `requires_empty_line_between_stub_file_statements` (not just preserving, enforcing an empty line).

Yeah, I'm going with `is_blank_line_required_after_nested_stub_class`.

Reply:
> Yeah, I'm going with `is_blank_line_required_after_nested_stub_class`.

Actually, with your other comment, I'm thinking of `should_insert_blank_line_after_class_in_stub_file`.